inspire/base/templates/records/Inspire_Default_HTML_detailed_macros.tpl

```diff
@@ -0,0 +1,176 @@
+          <span class="currect-collection">Jobs</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% endif %}
+    {% endfor %}
+    {% if not collection_found %}
+      <span>Literature</span>
+      <span>Authors</span>
+      <span>Conferences</span>
+      <span>Jobs</span>
+    {% endif %}
+  {% endif %}
+{% endmacro %}
+
+{% macro display_current_collection() %}
+  {% if record.get('collections') %}
+  {% set collection_found = [] %}
+    {% set collections = record.collections %}
+    {% for collection in collections %}
+      {% set collection_name = collection.get('primary') %}
+      {% if collection_name == "HEP" %}
+        {% if not collection_found %}
+          <span class="currect-collection">Literature</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% elif collection_name == "HEPNAMES" %}
+        {% if not collection_found %}
+          <span class="currect-collection">Authors</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% elif collection_name == "INSTITUTION" %}
+        {% if not collection_found %}
+          <span class="currect-collection">Institutions</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% elif collection_name == "CONFERENCES" %}
+        {% if not collection_found %}
+          <span class="currect-collection">Conferences</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% elif collection_name == "JOB" %}
+        {% if not collection_found %}
+          <span class="currect-collection">Jobs</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% elif collection_name == "EXPERIMENT" %}
+        {% if not collection_found %} 
+          <span class="currect-collection">Experiments</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% elif collection_name == "JOURNALS" %}
+        {% if not collection_found %}
+          <span class="currect-collection">Journals</span>
+          {% do collection_found.append(1) %}
+        {% endif %}
+      {% endif %}
+    {% endfor %}
+  {% endif %}  
+{% endmacro %}
+
+{% macro search_bar() %}
+  <div class="search-box pull-right" style="display: inline-block;">
+    <form class="search-form" action="/search">
+      <input type="text" name="p" placeholder=" {{ current_collection() | safe | trim }}" value="">
+        {% if record.get('collections') %}
+          {% set collections = record.collections %}
+          {% for collection in collections %}
+            {% set collection_name = collection.get('primary') %}
+            {% if collection_name == "HEP" %}
+              <input type="hidden" name="cc" value="{{ collection_name }}">
+            {% endif %}
+          {% endfor %}
+        {% endif %}
+    </form>
+  </div>
+{% endmacro %}
+
+{% macro current_collection() %}
+  {% if record.get('collections') %}
```

Comment:
All these macros related to collections seem quite similar, is there a way to refactor into a single macro?